dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java

```diff
@@ -1618,6 +1644,14 @@ private ImportSummary saveEvent( Program program, ProgramInstance programInstanc
             return importSummary.incrementIgnored();
         }
 
+        if ( event.getAssignedUser() != null )
+        {
+            if ( programStageInstance.getProgramStage().isEnableUserAssignment() )
+            {
+                programStageInstance.setAssignedUser( assignedUser );
```

Comment:
If assignedUser was already part of event object, why sending assignedUser as an additional parameter here? The check is done for event.getAssignedUser() in the if construct above but you finally end up using assignedUser parameter.

Reply:
I wanted to follow the same structure as the rest of the methods, where other refrences are fetched and passed as parameters . event.assignedUser = UID, while psi.assignedUser = User